pkg/receive/multitsdb.go

```diff
@@ -106,6 +106,14 @@ func (t *MultiTSDB) SkipMatchExternalLabels() {
 	t.skipMatchExternalLabels = true
 }
 
+func (t *MultiTSDB) GetTenants() map[string]*tenant {
+	return t.tenants
```

Comment:
This map is protected by `mtx`. A call site of `GetTenants()` can hold a pointer to the map and do reads/writes while the map is being updated by other goroutines. That'd be a data race.
It's safer to return a deep copy of the map if copying is not too expensive.

Reply:
refactored code